StripeConnect/StripeConnect/Source/Internal/Webview/HTTPStatusError.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  HTTPStatusError.swift
+//  
+//
+//  Created by Mel Ludowise on 10/14/24.
+//
+
+@_spi(STP) import StripeCore
+
+/// Error passed to the when the `ConnectComponentWebViewController.didFailLoadWithError`
+/// when receiving an error status code loading the component web page
+struct HTTPStatusError: Error, CustomNSError {
+    /// The HTTP status code
+    let errorCode: Int
+}
+
+extension HTTPStatusError: LocalizedError {
+    var errorDescription: String? {
+        NSError.stp_unexpectedErrorMessage()
+    }
+}
```

Comment:
Is there any value in having a localized error message? The only thing I can think of is that if the app decides displays the error message to the user that it's at least localized and not super technical. 

If we decide to keep this then I think I should also add it to `EmbeddedComponentError` since its debug description isn't localized.

Reply:
I would assume that integrators would appreciate the convenience of being able to directly use a user facing error message from the SDK. We could add a task for this.